packages/apps/plugins/plugin-deck/src/components/DeckLayout.tsx

```diff
@@ -573,15 +587,95 @@ export const DeckLayout = ({
                           <Plank.ResizeHandle classNames='row-start-[toolbar-start] row-end-[content-end]' />
                         </div>
                       ) : (
-                        !soloMain && <Plank.ResizeHandle classNames='row-span-3' />
+                        <Plank.ResizeHandle classNames='row-span-3' />
                       )}
                     </Plank.Root>
                   );
                 })}
               </Deck.Root>
             </div>
           </Main.Content>
-        ) : (
+        )}
+
+        {/* Solo main content surface. */}
+        {layoutMode === 'solo' && layoutParts.solo && layoutParts.solo.length > 0 && (
+          <Main.Content bounce classNames={['grid', 'block-end-[--statusbar-size]']}>
+            <div role='none' className='relative'>
+              <Deck.Root
+                overscroll={overscroll === 'centering'}
+                classNames={mx(
+                  'absolute inset-0',
+                  !flatDeck && 'surface-deck',
+                  slots?.wallpaper?.classNames,
+                  slots?.deck?.classNames,
+                )}
+                solo={true}
+              >
+                {layoutParts.solo.map((layoutEntry, index) => {
```

Comment:
wondering if this should only take the first element considering its supposed to be "solo"?

Reply:
There's some logic so that the array only ever contains 0 / 1 element. Doing the map is the simplest thing here in JSX I think.